provisioner/provisioner.go

```diff
@@ -555,33 +557,39 @@ func (p *Provisioner) executeCnspec(ui packer.Ui, comm packer.Communicator) erro
 
 		ui.Message("scan packer build")
 		scanService := scan.NewLocalScanner(scannerOpts...)
-		result, err = scanService.Run(context.Background(), scanJob)
+		res, err = scanService.Run(context.Background(), scanJob)
 		if err != nil {
 			ui.Error("scan failed: " + err.Error())
 			return err
 		}
 	}
 
+	report := res.GetFull()
 	ui.Message("scan completed successfully")
 
 	// render terminal output
+	handlerConf := reporter.HandlerConfig{
+		Format:       p.config.Output,
+		OutputTarget: p.config.OutputTarget,
+		Incognito:    p.config.Incognito,
+	}
+	outputHandler, err := reporter.NewOutputHandler(handlerConf)
+	if err != nil {
+		ui.Error("failed to create an output handler: " + err.Error())
+	}
+
 	buf := &bytes.Buffer{}
-	output := p.config.Output
-	format := reporter.Formats[output]
-	r := reporter.NewReporter(format, p.config.Incognito).WithOutput(buf)
+	if x, ok := outputHandler.(*reporter.Reporter); ok {
```

Comment:
nit: why not just create the `reporter.Reporter` directly instead of going through the interface creation? that will save the cast conversion here

Reply:
oh, i see what this code is doing, you can ignore this comment